service/r2/kv/store.go

```diff
@@ -73,6 +74,15 @@ func (s *store) FetchNamespaces() (*rules.NamespacesView, error) {
 	}, nil
 }
 
+func (s *store) ValidateRuleSet(rs *rules.RuleSetSnapshot) error {
+	validator := s.opts.Validator()
+	// If no validator is set, then the validation functionality is not applicable
+	if validator == nil {
+		return errors.New("no validator set on StoreOptions so validation is not applicable")
```

Comment:
You can use fmt.Errorf here dont need to mess with imports.

Reply:
The general rule of thumb is actually to only use `fmt.Errorf` when you need to format some values as part of the error message. For errors that are just a const string, `errors.New` is preferred.

Reply:
nit: make this a `errNilValidator = errors.New("no validator set on StoreOptions so validation is not applicable")`